backend/src/Edna/Dashboard/Service.hs

```diff
@@ -64,10 +73,53 @@ setIsSuspiciousSubExperiment subExpId isSuspicious = do
 -- primary.
 deleteSubExperiment :: SubExperimentId -> Edna NoContent
 deleteSubExperiment subExpId = NoContent <$ do
+  -- Check existence (not the optimal way, but it likely doesn't matter)
+  void $ getMeasurements subExpId
   logMessage $ fmt $ "Deleting sub-experiment " +| subExpId |+ ""
   unlessM (Q.deleteSubExperiment subExpId) $
     throwM $ DECantDeletePrimary subExpId
 
+-- | Create a new sub-experiment based on existing one.
+newSubExperiment ::
+  SubExperimentId -> NewSubExperimentReq -> Edna (WithId 'SubExperimentId SubExperimentResp)
+newSubExperiment subExpId req = do
+  (removed, newResult) <- analyseNewSubExperiment subExpId req
+  transact $ do
+    result <- subExperimentRecToResp <$>
+      Q.cloneSubExperiment subExpId (nserName req) newResult
+    result <$ insertRemovedMeasurements (wiId result) removed
```

Comment:
Looks like transact is not needed

Reply:
I think we should `cloneSubExperiment` and `insertRemovedMeasurements` atomically. If we insert sub-experiment and don't insert removed measurements (e. g. because the app crashes), we will get invalid data in DB.

From my experiments `transact` doesn't work like that, but it's a topic of another issue (EDNA-76). AFAIU it is supposed to give us atomicity, but doesn't give for some reason. Anyway, even if `transact` does nothing, it's a marker saying "these things should happen atomically".